pydal/migrator.py

```diff
@@ -429,7 +429,7 @@ def migrate_table(self, table, sql_fields, sql_fields_old, sql_fields_aux,
                     drop_expr % (table._rname, key_tmp)
                 ]
                 metadata_change = True
-            elif sql_fields[key]['type'] != sql_fields_old[key]['type']:
```

Comment:
Yes. If the metadata are still not identical at this point, the field can either be safely fake-migrated or it needs to be fixed manually anyway. Or do you prefer triggering migrations which do nothing on every single Web2py request that uses the table?